usom/operations.py

```diff
@@ -0,0 +1,78 @@
+from connectors.core.connector import Connector, ConnectorError, get_logger
+import requests
+import json
+import xmltodict
+import datetime
+import os.path
+import publicsuffix2
+import re
+import dateutil.parser
+import validators
+
+TMP_FILE = '/tmp/usom.cache.json'
+logger = get_logger('usom')
+
+
+def update(config):
+    try:
+        url = config.get("url")
+        verify = config.get("verify")
+        now = datetime.datetime.now()
+        if os.path.exists(TMP_FILE):
+            f = open(TMP_FILE)
+            content = json.load(f)
+            f.close()
+            old = dateutil.parser.isoparse(content['date'])
+            if (now - old) < datetime.timedelta(hours=config["expire_hours"]):
+                return False
+        try:
+            response = requests.get(url, verify=verify)
+            content = xmltodict.parse(response.text)
+            content = content['usom-data']['url-list']
+            content['date'] = now.isoformat()
+            out_file = open(TMP_FILE, "w")
+            json.dump(content, out_file, indent=6)
+            out_file.close()
+            return True
+        except requests.exceptions.RequestException as ex:
+            if response.status_code != 200:
```

Comment:
status_code != 200 won't raise an exception, you will need to handle it within 'try', check this out: https://stackoverflow.com/questions/16511337/correct-way-to-try-except-using-python-requests-module